controllers/metal3remediation_controller.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	//"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	capm3 "github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha4"
+	"github.com/metal3-io/cluster-api-provider-metal3/baremetal"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Metal3RemediationReconciler reconciles a Metal3Remediation object
+type Metal3RemediationReconciler struct {
+	client.Client
+	ManagerFactory baremetal.ManagerFactoryInterface
+	Log            logr.Logger
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=metal3remediations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=metal3remediations/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=machine,verbs=verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=machine/status,verbs=get;update;patch
+
+func (r *Metal3RemediationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	remediationLog := r.Log.WithValues("metal3remediation", req.NamespacedName)
+
+	// Fetch the Metal3Remediation instance.
+	metal3Remediation := &capm3.Metal3Remediation{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, metal3Remediation); err != nil {
+		if apierrors.IsNotFound(err) {
+			remediationLog.Error(err, "unable to get metal3Remediation")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	helper, err := patch.NewHelper(metal3Remediation, r.Client)
+	if err != nil {
+		remediationLog.Error(err, "failed to init patch helper")
+		return ctrl.Result{}, errors.Wrap(err, "failed to init patch helper")
```

Comment:
I think we can just return the `err` as is.
The err will be used to trigger a requeue, and we already logged the detailed message. 
```suggestion
		remediationLog.Error(err, "failed to init patch helper")
		return ctrl.Result{}, err
```